core/trino-main/src/main/java/io/trino/security/AccessControlUtil.java

```diff
@@ -43,19 +40,14 @@ public static List<BasicQueryInfo> filterQueries(Identity identity, List<BasicQu
                 .map(BasicQueryInfo::getSession)
                 .map(SessionRepresentation::toIdentity)
                 .filter(owner -> !owner.getUser().equals(identity.getUser()))
-                .map(FullIdentityEquality::new)
                 .distinct()
-                .map(FullIdentityEquality::getIdentity)
                 .collect(toImmutableList());
-        owners = accessControl.filterQueriesOwnedBy(identity, owners);
+        Collection<Identity> allowedOwners = accessControl.filterQueriesOwnedBy(identity, owners);
 
-        Set<FullIdentityEquality> allowedOwners = owners.stream()
-                .map(FullIdentityEquality::new)
-                .collect(toImmutableSet());
         return queries.stream()
                 .filter(queryInfo -> {
                     Identity queryIdentity = queryInfo.getSession().toIdentity();
-                    return queryIdentity.getUser().equals(identity.getUser()) || allowedOwners.contains(new FullIdentityEquality(queryIdentity));
+                    return queryIdentity.getUser().equals(identity.getUser()) || allowedOwners.contains(queryIdentity);
```

Comment:
nit: previously it was a set whereas now it's just a collection so contains might be a bit slower (assuming the returned collection is not actually a set ofc)